### PR TITLE
[FIX] product_template : Fix inconsistency on the variant choice

### PR DESCRIPTION
**Pull request by qni-odoo**
On e-commerce, when clicking on one product, the system does not always pick the same variant for the product.
This may lead to inconsistent results, as variant details differ.

We fix this by making the ordering total.

opw-2340958

### CODE REVIEW

**Comment by qni-odoo** [comment]
@rim-odoo 
Here is the pr for the inconsistency of the product on the ECommerce

**Comment by rim-odoo** [comment]
I would rephrase the commit message like this:
```
[FIX] product: inconsistency in product variant selection

On e-commerce, when clicking on one product, the system does not always pick the same variant for the product.
This may lead to inconsistent results, as variant details differ.

We fix this by making the ordering total.

opw-2340958
```

**Comment by rim-odoo** [comment]
@qni-odoo Can you fix the commit message, then you will be able to r-plus the branch 

@robodoo delegate+

**Review by rim-odoo — APPROVED** [review]

**Comment by qni-odoo** [comment]
@robodoo r+

**Comment by rim-odoo** [comment]
@fw-bot up to saas-12.3
